sdks/python/apache_beam/io/filesystems.py

```diff
@@ -35,28 +38,38 @@
 # pylint: disable=wrong-import-position, unused-import
 try:
   from apache_beam.io.hadoopfilesystem import HadoopFileSystem
-except ImportError:
+except ModuleNotFoundError:
```

Comment:
In which scenario do we expect ModuleNotFound error?

Reply:
Well, of course, when `apache_beam.io.hadoopfilesystem.HadoopFileSystem` is not installed... Wait, the Filesystem is under `apache_beam.io`, so the module is embedded by default Apache Beam?? If so `LocalFileSystem` below should also be installed by default, so this kind of handling would be unnecessary.

Just to confirm, I would like a Yes/No answer!

Reply:
It's No, not necessary and not working, based on my testing

I tested that if I installed beam without hdfs, and run
```
from apache_beam.io.hadoopfilesystem import HadoopFileSystem
```

I will get

"Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/yathu/dev/virtualenv/py38beam/lib/python3.8/site-packages/apache_beam/io/hadoopfilesystem.py", line 29, in <module>
    import hdfs
ModuleNotFoundError: No module named 'hdfs'
"

However, ModuleNotFoundError is a subclass of ImportError. The current logic actually will make the warning get missed when trying to load HadoopFileSystem.

Reply:
Now I see,

when the optional dependencies not installed -> ModuleNotFoundError -> No error as before

when the dependency is installed, but persumably not installed correctly, some other ImportError (not ModuleNotFoundError), raise a warning

Reply:
sorry for my BIG late response. 
I'll take care of this PR this weekend!